test/puppetlabs/puppetdb/cli/services_test.clj

```diff
@@ -133,3 +123,47 @@
         (testing (format "%s requests are refused" (name v)))
         (is (retirement-response? v (ping v)))))))
 
+(deftest whitelist-middleware
+  (testing "should log on reject"
+    (let [wl (fs/temp-file "whitelist-log-reject")]
+      (.deleteOnExit wl)
```

Comment:
Similarly to my comment below we have https://github.com/puppetlabs/puppetdb/blob/1a226695db6ed96a704173b11ab9c61d53f340ef/test/puppetlabs/puppetdb/testutils.clj#L307-L317